tv_apps/fling-player/js/fling_player.js

```diff
@@ -164,79 +186,176 @@
+  };
+
+  proto._startAutoSeek = function (dir) {
 
     if (this._autoSeekStartTime == null) { // Do not double start
 
+      // To stop this is important.
+      // We don't want they get messed with each other
+      this._stopAutoUpdateControlPanel();
+
       this._autoSeekStartTime = (new Date()).getTime();
       this._autoSeekDirection = dir;
       this._autoSeek();
     }
   };
 
-  proto._stopAutoSeek = function fp_stopAutoSeek(dir) {
+  proto._stopAutoSeek = function (dir) {
     this._autoSeekStartTime = null;
     this._autoSeekDirection = null;
   };
 
-  proto._autoSeek = function fp_autoSeek() {
+  /**
+   * This is to handle this case that user seeks on video by long pressing key.
+   * The seeking policy would go based on duration of pressing
+   */
+  proto._autoSeek = function () {
 
     if (this._autoSeekStartTime != null) {
 
-      var current = this._player.getVideo().currentTime;
+      var time = this._player.getVideoCurrentTime();
       var factor = (this._autoSeekDirection == 'backward') ? -1 : 1;
       var seekDuration = (new Date()).getTime() - this._autoSeekStartTime;
       var seekStep = (seekDuration > this.AUTO_SEEK_LONG_PRESSED_SEC) ?
               this.AUTO_SEEK_STEP_LARGE_SEC : this.AUTO_SEEK_STEP_NORMAL_SEC;
 
-      current += factor * seekStep;
+      time += factor * seekStep;
+      time = Math.min(Math.max(time, 0), this._player.getVideoLength());
 
-      this._player.seek(current);
-      this.moveTimeBar('elapsed', current);
-      // TODO: Update time info
+      this.seek(time);
 
       setTimeout(this._autoSeek.bind(this), this.AUTO_SEEK_INTERVAL_MS);
     }
   };
 
+  proto.seek = function (sec) {
+    this._player.seek(sec);
+    this.moveTimeBar('elapsed', sec);
+    this.writeTimeInfo('elapsed', sec);
+    this.showControlPanel(true);
+  };
+
   // Video handling end
 
   // Event handling
-  proto.handleEvent = function fp_handleEvent(e) {
+
+  proto.handleEvent = function handleVideoEvent(e) {
 
     mDBG.log('FlingPlayer#handleEvent: e.type = ' + e.type);
 
-    var data = { 'time': this._player.currentTime };
+    var data = { 'time': this._player.getVideoCurrentTime() };
 
     switch (e.type) {
 
-      case 'waiting':
-        this.setLoading(true);
-        this._connector.reportStatus('buffering', data);
-      break;
-
       case 'loadedmetadata':
+        this.writeTimeInfo('elapsed', this._player.getVideoCurrentTime());
+        this.writeTimeInfo('duration', this._player.getVideoLength());
         this._connector.reportStatus('loaded', data);
       break;
 
+      case 'waiting':
+        this.showLoading(true);
+        this._connector.reportStatus('buffering', data);
+      break;
+
       case 'playing':
-        this.setLoading(false);
+        this.showLoading(false);
+        // TODO: Hide 'Starting video cast from ...'
         this._connector.reportStatus('buffered', data);
+
+        this._startAutoUpdateControlPanel();
         this.showControlPanel(true);
```

Comment:
Let UI controlled by the same function. move `showControlPanel` to the function you updating control panel.